1144 indicator for detail

dstopp opened the pull request:
- Fixes #1144 
- Were any Django, import pipeline, table_schema, or dictionary files modified? Y/N
  - If YES:
    - Database used: opus3_test_210526
    - All Django tests pass: Y
    - FLAKE8 run on modified code: N
- Were any JavaScript or CSS files modified? Y
  - If YES:
    - JSHINT run on all affected files: Y
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: Y
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): Y

Description of changes:
-add 'DETAIL' tag to current selected detail opus object
-add mini-icon/image to the Detail tab to show detail has a selected value; hover to get opus id
-fix the regression tests for windows delete

Known problems:

dstopp committed:
Add indicator on gallery view and NAV of what OPUSID is on detail tab

-updates browse and cart tabs with the word "DETAIL" for currently selected detail item
-adds the OPUS_ID to the nav, if there is one for detail

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1144_indicator_for_detail

dstopp committed:
Updated detail tab to show small thumbnail of current detail view

dstopp committed:
update the test helper code for windows

rfrenchseti commented on opus/application/static_media/js/detail.js:
```diff
@@ -128,6 +128,23 @@ var o_detail = {
+    showDetailThumbInNav: function(imageHtml) {
+        if (imageHtml === undefined) {
+            if (opus.prefs.detail != "") {
+                let url = "/opus/api/image/thumb/" + opus.prefs.detail + ".json";
```
This should use the internal API URL "__api/image/thumb/" so that we don't count these as uses by an external API user. We haven't needed this before so you'll have to add it to urls.py by just copying the normal "api/image" line and adding a new one with the "__" in front. Everything else can stay the same. The internal functionality doesn't change - we just use it when parsing the Apache logs.

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -1242,6 +1242,15 @@ var o_browse = {
         o_browse.hideMenus();
+
+        // if the item is selected as 'detail', add the icon in the corner
+        // hide all first... this will hide both on browse and cart
```
Hide all what first? And why are you hiding anything? This comment is confusing. And what "icon" in the corner? Do you mean the word "detail"?

dstopp committed:
Address pull request comments

-use internal API to get the image url
-update/clarify the comment regarding the "DETAIL" designation

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -1242,6 +1242,15 @@ var o_browse = {
+
+        // if the item is selected as 'detail', add the icon in the corner
+        // hide all first... this will hide both on browse and cart
+        $(`.op-thumbnail-container .op-detail-overlay`).addClass("op-hide-element");
```
Please change `` to "" since there is no string interpolation here.

dstopp committed:
change quote types to be consistent w/coding standard
